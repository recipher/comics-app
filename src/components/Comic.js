import React, { memo } from 'react';

const Comic = memo(({ title, description, image }) => {
  return (
    <div>
      <h3>{title}</h3>
      <p dangerouslySetInnerHTML={{ __html: description }}/>
      <img src={image} alt={description} />
      <hr/>
    </div>
  );
});

export default Comic;
