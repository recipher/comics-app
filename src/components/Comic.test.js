import React from 'react';
import 'react-testing-library/cleanup-after-each';
import { render } from 'react-testing-library';
import Comic from './Comic';

const COMIC = {
  id: 1234,
  title: 'Captain Marvel',
  description: '...',
  image: 'image.jpg',
};

it('should match snapshot', () => {
  const { container } = render(<Comic {...COMIC} />);

  expect(container).toMatchSnapshot();
});
