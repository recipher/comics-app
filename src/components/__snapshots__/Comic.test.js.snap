// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div>
    <h3>
      Captain Marvel
    </h3>
    <p>
      ...
    </p>
    <img
      alt="..."
      src="image.jpg"
    />
    <hr />
  </div>
</div>
`;
